src/veranda/raster/native/geotiff.py

```diff
@@ -0,0 +1,498 @@
+                                                        nodataval=nodataval,
+                                                        scale_factor=scale_factor,
+                                                        offset=offset,
+                                                        dtype=dtype,
+                                                        **encoder_kwargs),
+                                                        xoff=col, yoff=row)
+            else:
+                self.src.GetRasterBand(band).WriteArray(data_write[band], xoff=col, yoff=row)
+
+    def __to_dict(self, arg):
+        """
+        Assigns non-iterable object to a dictionary with band numbers as keys. If `arg` is already a dictionary the
+        same object is returned.
+
+        Parameters
+        ----------
+        arg : non-iterable or dict
+            Non-iterable, which should be converted to a dict.
+
+        Returns
+        -------
+        arg_dict : dict
+            Dictionary mapping band numbers with the value of `arg`.
+
+        """
+        if not isinstance(arg, dict):
+            arg_dict = dict()
+            for band in self.bands:
+                arg_dict[band] = arg
+        else:
+            arg_dict = arg
+
+        return arg_dict
+
+    def flush(self):
+        """
+        Flush data on disk.
+        """
+        if self.src is not None:
+            self.src.FlushCache()
+
+    def close(self):
+        """
+        Close the dataset.
+        """
+        self.src = None
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        self.close()
+
+
+def create_vrt_file(filepaths, vrt_filepath, shape, sref_wkt, geotrans, bands=1):
+    """
+    Creates a VRT file stack from a list of file paths.
+
+    Parameters
+    ----------
+    filepaths : list of str
+        Full system path to the files to stack.
+    vrt_filepath : str
+        Full system path to the VRT file to create.
+    shape : 2-tuple
+        Shape (rows, columns) of the raster stack.
+    sref_wkt : str
+        Coordinate Reference System (CRS) information in WKT format.
+    geotrans : 6-tuple or list, optional
+        Geo-transformation parameters with the following entries:
+            0: Top-left X coordinate
+            1: W-E pixel sampling
+            2: Rotation, 0 if image is axis-parallel
+            3: Top-left Y coordinate
+            4: Rotation, 0 if image is axis-parallel
+            5: N-S pixel sampling (negative value if North-up)
+    bands : tuple or list or int, optional
+        Band number(s). Defaults to 1.
+
+    """
+    n_filepaths = len(filepaths)
+    bands = to_list(bands)
+    n_bands = len(bands)
+    n_rows, n_cols = shape
+
+    ref_filepath = filepaths[0]
+    band_attr_dict = dict()
+    band_attr_dict['nodataval'] = []
+    band_attr_dict['scale_factor'] = []
+    band_attr_dict['offset'] = []
+    band_attr_dict['dtype'] = []
+    band_attr_dict['blocksize'] = []
```

Comment:
you can shorten this to 1 line with a default dictionary:
```
band_attr_dict = defaultdict(list)
```

Reply:
Very cool, did not know about that!